src/rust/engine/fs/store/src/remote.rs

```diff
@@ -39,6 +39,35 @@ impl fmt::Debug for ByteStore {
   }
 }
 
+/// Represents an error from accessing a remote bytestore.
+pub enum ByteStoreError {
+  /// gRPC error
+  Grpc(Status),
+
+  /// Other errors
+  Other(String),
+}
+
+impl fmt::Display for ByteStoreError {
+  fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+    match self {
+      ByteStoreError::Grpc(status) => fmt::Display::fmt(status, f),
+      ByteStoreError::Other(msg) => fmt::Display::fmt(msg, f),
+    }
+  }
+}
+
+impl fmt::Debug for ByteStoreError {
```

Comment:
Why not just derive this? Being able to differentiate between Grpc/non seems like it may be useful?

Reply:
I just copied the impl for `fmt::Display` and did `s/Display/Debug`. Will switch to a derive. The error messages will be different enough and will differentiate them, no need to append a "gRPC error:" or other tag in front.